client/src/Datasets/datasets.hooks.ts

```diff
@@ -0,0 +1,151 @@
+import { useState, useEffect, Dispatch, SetStateAction } from 'react';
+import { useNavigation } from 'react-navi';
+import { useFormik, FormikProps } from 'formik';
+import * as Yup from 'yup';
+
+import { Dataset } from '../common/types';
+import * as DatasetApi from './datasets.api';
+import useGlobalContext from '../common/hooks/useGlobalContext';
+import { addProcess, removeProcess } from '../common/utils/globalActions';
+
+// interfaces
+interface DatasetsState {
+    list: ListState;
+    dialogs: DialogsState;
+    navigation: NavigationState;
+}
+interface NavigationState {
+    openDetails(dataset: Dataset): void;
+}
+interface ListState {
+    page: number;
+    pageCount: number;
+    datasets: Dataset[];
+    categories: string[];
+    setPage: Dispatch<SetStateAction<number>>;
+    refreshPage(): void;
+}
+interface DialogsState {
+    help: [boolean, Dispatch<SetStateAction<boolean>>];
+    create: [boolean, Dispatch<SetStateAction<boolean>>];
+    share: [boolean, Dispatch<SetStateAction<boolean>>];
+    edit: [boolean, Dispatch<SetStateAction<boolean>>];
+}
+interface FormCreateState {
+    name: string;
+    categories: Array<string>;
+}
+
+// hooks
+export const useDatasetsPage = (): DatasetsState => {
+    const navigation = usePageNavigation();
+    const list = useList(52);
+    const dialogs = useDialogs();
+
+    return {
+        list: list,
+        dialogs: dialogs,
+        navigation: navigation,
+    };
+};
+
+const usePageNavigation = (): NavigationState => {
+    const navigation = useNavigation();
+    const openDetails = (dataset: Dataset) => {
+        navigation.navigate(`/dataset/${dataset.id}`);
+    };
+    return {
+        openDetails: openDetails,
+    };
+};
+
+const useList = (itemsPerPage: number): ListState => {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [_globalState, dispatch] = useGlobalContext();
```

Comment:
you should be able to replace the `_globalState` with just `_` and eslint is going to play nicely